opm/simulators/linalg/gpuistl/GpuOwnerOverlapCopy.hpp

```diff
@@ -275,24 +285,25 @@ class GPUAwareMPISender : public GPUSender<field_type, OwnerOverlapCopyCommunica
 
     void buildCommPairIdxs() const
     {
-        auto &ri = this->m_cpuOwnerOverlapCopy.remoteIndices();
+        const auto& ri = this->m_cpuOwnerOverlapCopy.remoteIndices();
         std::vector<int> commpairIndicesCopyOnCPU;
         std::vector<int> commpairIndicesOwnerCPU;
 
-        for(auto process : ri) {
+        for (auto process : ri) {
```

Comment:
I don't off hand know the size of `ri` here, but this range-`for` will copy the full sequence.  If you're changing this anyway, maybe we could rewrite it as
```C++
for (const auto& process : ri) {
    // ...
}
```
instead?

Reply:
good idea.